schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/dao/common/ActivityPackageActivityDao.java

```diff
@@ -0,0 +1,49 @@
+package com.jasify.schedule.appengine.dao.common;
+
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
+import com.jasify.schedule.appengine.dao.BaseCachingDao;
+import com.jasify.schedule.appengine.dao.BaseDaoQuery;
+import com.jasify.schedule.appengine.meta.activity.ActivityPackageActivityMeta;
+import com.jasify.schedule.appengine.model.EntityNotFoundException;
+import com.jasify.schedule.appengine.model.activity.Activity;
+import com.jasify.schedule.appengine.model.activity.ActivityPackageActivity;
+import com.jasify.schedule.appengine.model.activity.ActivityType;
+import org.slim3.datastore.CompositeCriterion;
+import org.slim3.datastore.Datastore;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author szarmawa
+ * @since 09/06/15.
+ */
+public class ActivityPackageActivityDao extends BaseCachingDao<ActivityPackageActivity> {
+    private final ActivityTypeDao activityTypeDao = new ActivityTypeDao();
+
+    public ActivityPackageActivityDao() {
+        super(ActivityPackageActivityMeta.get());
+    }
+
+    public List<ActivityPackageActivity> getBy(final Activity activity) {
+        ActivityPackageActivityMeta meta = getMeta();
+        try {
+            ActivityType activityType = activityTypeDao.get(activity.getActivityTypeRef().getKey());
```

Comment:
1. Warning:  `activity.getActivityTypeRef().getKey()` might be **null**
2. Warning: if someone deleted the activityType, then `activityTypeDao.get(...)` will throw EntityNotFoundException